### PR TITLE
Fix compilation + event v4 update

### PR DESCRIPTION
**Pull request by Juuxel**
# compilation
- "Logging" before constructor super calls broke compilation in general
- The `TargetViolator` broke compilation with Gradle
# event
- It now warns you if EventManager.register was called without actually executing the registration
- Support for MenuTransformers in OpenMenuEvent
- IO.merge supports custom types as well
- OpenMenuEvent actually takes the *opened* menu instead of the player menu now
- Fixed crashes with no listeners for an event

### CODE REVIEW

File: src/main/java/io/github/fabriccommunity/everything/api/elegant/position/Offset.java [hunk 1]
```diff
@@ -40,12 +40,6 @@ public Offset(final BlockPos pos, final Direction direction, final int amount) {
     }
 
     private Offset(final int x, final int y, final int z) {
-        System.out.println(" +\"\"\"\"\"+ ");
-        System.out.println("[| o o |]");
-        System.out.println(" |  ^  | ");
-        System.out.println(" | '-' | ");
-        System.out.println(" +-----+ ");
-        System.out.println("fran is logging messages! call the antivirus!");
```
Comment (valoeghese) [inline src/main/java/io/github/fabriccommunity/everything/api/elegant/position/Offset.java:48]: ah yes

File: src/main/java/io/github/fabriccommunity/everything/api/event/v4/EventManager.java [hunk 2]
```diff
@@ -90,20 +92,43 @@
 
             @Override
             public IO<Boolean> register(final EventHandler<A> handler) {
+                final class ExecutionTrackingIO<A> implements IO<A> {
```
Comment (valoeghese) [inline src/main/java/io/github/fabriccommunity/everything/api/event/v4/EventManager.java:95]: wait is this even legal @Juuxel

File: src/main/java/io/github/fabriccommunity/everything/api/elegant/position/Offset.java (hunk 1 above)
Comment (franiscoder) [reply src/main/java/io/github/fabriccommunity/everything/api/elegant/position/Offset.java:48]: i dissaprove of this!!!!!

**Review by frqnny — CHANGES_REQUESTED** [review]

**Review by valoeghese — COMMENTED** [review]
e

**Comment by franiscoder** [comment]
add back the very important logging messages or else

**Comment by Juuxel** [comment]
@franiscoder your "logging" makes it impossible to compile everything api 😡

File: src/main/java/io/github/fabriccommunity/everything/api/event/v4/EventManager.java (hunk 2 above)
Comment (Juuxel) [reply src/main/java/io/github/fabriccommunity/everything/api/event/v4/EventManager.java:95]: Yes. Local classes are fun :^)

**Comment by frqnny** [comment]
yes what is the problem with editing everything API with a web browser smh

**Pull request merged**